src/auth/__tests__/authService.test.js

```diff
@@ -269,18 +283,87 @@ describe('authService', () => {
     it('should call sessionStorage.setItem with the right arguments', async () => {
       const setSpy = jest.spyOn(Storage.prototype, 'setItem');
       expect.assertions(2);
-      await authService.fetchApiToken(mockUser);
+      await authService.fetchAndStoreApiToken(mockUser);
 
       expect(setSpy).toHaveBeenCalledTimes(1);
       expect(setSpy.mock.calls[0]).toMatchInlineSnapshot(`
         Array [
           "apiToken",
-          "71ffd52c-5985-46d3-b445-490554f4012a",
+          "${apiToken}",
         ]
       `);
     });
   });
 
+  describe(`Api tokens are fetched again after user tokens are renewed.
+            After silent renew completes, the _userLoaded event is raised and`, () => {
+    beforeEach(() => {
+      global.fetch.resetMocks();
+      mockFetchApiToken();
+      jest.useFakeTimers();
+    });
+
+    afterEach(() => {
+      jest.useRealTimers();
+    });
+    beforeAll(() => {
+      enableActualHttpPoller(jest.requireActual('../http-poller'));
+    });
+
+    it('it calls fetchApiToken and stores new token after success', async () => {
+      const { sessionUser } = setSession({
+        validUser: true,
+        validApiToken: true,
+      });
+      sessionStorage.setItem(API_TOKEN, 'foobar');
```

Comment:
I always like to give test data such contents that it helps figuring out what's wrong if the test shall ever fail. For example here instead of "foobar" this could be "old api token" or something like that. If `apiToken` was equal to "new api token", then if the final assert in this test would fail, an error message would be printed that could look something like this (completely made up example, I don't know how the used test framework really prints errors):

```
Expected `sessionStorage.getItem(API_TOKEN)` to be equal to `apiToken`.
    `sessionStorage.getItem(API_TOKEN)` == "old api token"
    `apiToken` == "new api token"
```

Reply:
This is a good point! Changed. 

The printed error shows assumed value and received value.